packages/server/src/fhir/sql.ts

```diff
@@ -613,6 +621,116 @@ export class SelectQuery extends BaseQuery implements Expression {
   }
 }
 
+export class SelectPerReferenceQuery extends BaseQuery implements Expression {
```

Comment:
Hm.  This mixes a lot of FHIR business logic with SQL builder, and one of the goals of SQL builder was to keep it general purpose.  How hard would it have been to use the normal `SelectQuery` for this?

Reply:
`SelectPerReferenceQuery` should be moved from `sql.ts` into `packages/server/src/fhir/search.ts` as an implementation detail since it is indeed not at all generic.

Reply:
Now that I have a better grasp on `SelectQuery` and friends, it probably wouldn't be too bad to rewrite `SelectPerReferenceQuery` as just a function if your concern is around structuring it as a class.